src/MLJClusteringInterface.jl

```diff
@@ -177,27 +177,36 @@ MMI.reporting_operations(::Type{<:DBSCAN}) = (:predict,)
     branchorder::Symbol = :r :: (_ ∈ (:r, :barjoseph, :optimal))
     h::Union{Nothing,Float64} = nothing
     k::Int = 3
-    _cache = nothing
 end
+"""
+    struct DendrogramCutter{T}
+        dendrogram::T
+    end
 
-function compute_dendrogram(X, metric, linkage, branchorder)
-#     @info "computing dendrogram"
-    Xarray = MMI.matrix(X)
-    d = pairwise(metric, Xarray, dims = 1) # n x n
-    Cl.hclust(d, linkage = linkage, branchorder = branchorder)
+Callable object to cut a dendrogram.
+"""
+struct DendrogramCutter{T}
+    dendrogram::T
+end
+"""
+    (cutter::DendrogramCutter)(; h = nothing, k = 3)
+
+Cuts the dendrogram at height `h` or, if `height == nothing`, such that `k` clusters are obtained.
+"""
+function (cutter::DendrogramCutter)(; h = nothing, k = 3)
+    MMI.categorical(Cl.cutree(cutter.dendrogram, k = k, h = h))
+end
+function Base.show(io::IO, ::DendrogramCutter)
+    println(io, "Dendrogram Cutter.")
```

Comment:
Shouldn't `println`  be just `print` here?
```suggestion
    print(io, "Dendrogram Cutter.")
```